src/uct/ib/mlx5/rc/rc_mlx5_common.c

```diff
@@ -620,40 +621,58 @@ uct_rc_mlx5_dp_ordering_ooo_init(uct_rc_mlx5_iface_common_t *iface,
      * - if dp_ordering_ooo is set, QPC/DCTC can enable AR.
      * - if dp_ordering_ooo_force is set, QPC/DCTC can request mlxreg
      *   configuration override, useful to force disable.
-     *
-     * QP modify behavior with returned values:
-     * - UCS_AUTO: Do not affect existing system behavior.
-     * - UCS_NO  : Force AR disabling on the QP if supported. QP modify will
-     *   return error on failure.
-     * - UCS_TRY : Set AR to enable, ignored if any failure.
-     * - UCS_YES : Force AR enabling on the QP if supported. QP modify will
-     *   return error on failure.
+     * 
+     * Fail at the following cases:
+     * - Want to force configuration but can't force it
+     * - Want to force enable or force disable adaptive routing but it is unavailable
+     * - Want to force disable adaptive routing but force enabling DDP
      */
+    iface->config.force_ordering = ucs_ternary_auto_value_is_yes_or_no(
+                                           config->super.ar_enable) ||
+                                   (config->ddp_enable == UCS_YES);
+
+    if ((iface->config.force_ordering && !dp_ordering_ooo_force) ||
+        ((config->super.ar_enable == UCS_NO) &&
+         (config->ddp_enable == UCS_YES))) {
```

Comment:
need a comment to explain this check, move the comment from lines 626-638 to here?